### PR TITLE
Move Conda Nightly builds to Docker

### PR DESCRIPTION
**Pull request by mathfac**
## Motivation and Context
https://github.com/facebookresearch/habitat-sim/pull/663 uses Ubuntu 16.04 as base OS to build conda builds. That limits conda build compatibility compare to using Centos7 as base OS.
- Added install Docker step to Nightly builds job
- Added "mesa-libGL-devel" install to conda Docker to improver build with Display
- Moved build conda env creation to docker build phase

## How Has This Been Tested
- Checked that nightly builds are working good on Colab

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -463,17 +463,27 @@ jobs:
...
       - run:
           name: Build conda Linux packages
+          no_output_timeout: 240m
           command: |
               export PATH=$HOME/miniconda/bin:$PATH
               . activate habitat;
               cd habitat-sim
 
+              # install Docker
+              curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
```
Comment (Skylion007) [inline .circleci/config.yml:473]: Any reason you aren't using the magic `- setup_remote_docker` per the [documentation](https://circleci.com/docs/2.0/building-docker-images/)?

Comment (mathfac) [reply .circleci/config.yml:473]: `setup_remote_docker: step is not compatible with machine executor. You can use Docker without a special step if the build image has it installed.` Looks, like using `setup_remote_docker ` you should use different executor other than we use now.

Comment (Skylion007) [reply .circleci/config.yml:473]: @mathfac "It is possible to specify a different executor type for every job in your ‘.circleci/config.yml’ by specifying the executor type and an appropriate image. An image is a packaged system that has the instructions for creating a running environment. A container or virtual machine is the term used for a running instance of an image. For example, you could specify an executor type and an image for every job:" https://circleci.com/docs/2.0/executor-types/

**Comment by Skylion007** [comment]
Building the conda viewer still doesn't work. :( If just building the headless conda binaries unblock us, that's fine with me. We can add the viewer back in a separate PR.

**Comment by mathfac** [comment]
We should be good to go here.
Screenshot 2020-07-14 14 10 35

**Pull request merged**